### DISCUSSION TITLE
Make defaultAlchemyApiKey obsolete

### DISCUSSION THREAD

**Discussion by 0xhaiku** [wevm/wagmi#1151] [Ideas]
It seems that there are times when the use of the default alchemy api returns an error, perhaps due to an increase in the number of users. The following could be due to a misconfiguration of api key, and I had the same situation. 

https://github.com/wagmi-dev/wagmi/issues/1133
https://github.com/wagmi-dev/wagmi/discussions/1146

Just a recommendation, but it might be better to return an error if the api key is not there, so you don't get extra issues. Thanks for the nice library.

**Comment by soccerarenadev** [comment]
I'm getting that error, but it is not due to too many request.
This is the actual message:
{"message":"Method Not Allowed","logref":null,"path":null,"_links":{"self":{"href":"/v2/asdfasdfasdf","templated":false,"profile":null,"deprecation":null,"title":null,"hreflang":null,"type":null,"name":null}},"_embedded":{"errors":[{"message":"Method [GET] not allowed for URI [/v2/asdfasdfasdf]. Allowed methods: [POST]","logref":null,"path":null,"_links":{},"_embedded":{}}]}}
More info: https://github.com/wagmi-dev/wagmi/issues/1133

**Comment by tmm** [comment]
Thanks for the thoughts! We [removed the default Alchemy ID from the public RPC URLs](https://github.com/wagmi-dev/wagmi/pull/1176) and [made RPC provider API keys required moving forward](https://github.com/wagmi-dev/wagmi/pull/1185) starting in version 0.8.
